torch/csrc/jit/passes/autocast.cpp

```diff
@@ -482,5 +502,58 @@ void Autocast(const std::shared_ptr<Graph>& graph) {
   GRAPH_DUMP("\nAfter Autocast: ", graph);
 }
 
+// Note: Initially we attempted to overload autocast functions with `Tensor`
+// input, but that turns out to be tricky for JIT, since overload doesn't work
+// well between `Tensor` & `Optional[Tensor]` in `Node::matches(FunctionSchema)`
+RegisterOperators reg_autocast_to_reduced_precision_optional({
+    Operator(
+        "aten::_autocast_to_reduced_precision_optional(Tensor(a)? self, bool cuda_enabled, bool cpu_enabled, ScalarType cuda_dtype, ScalarType cpu_dtype) -> Tensor(a)?",
+        [](const Node* node) -> Operation {
+          return [](Stack& stack) {
+            auto cpu_dtype = pop(stack).toScalarType();
+            auto cuda_dtype = pop(stack).toScalarType();
+            auto cpu_enabled = pop(stack).toBool();
+            auto cuda_enabled = pop(stack).toBool();
+            auto self = pop(stack).toOptional<at::Tensor>();
+
+            c10::optional<at::Tensor> ret = c10::nullopt;
+
+            if (self.has_value()) {
+              ret = at::native::_autocast_to_reduced_precision(
+                  self.value(),
+                  cuda_enabled,
+                  cpu_enabled,
+                  cuda_dtype,
+                  cpu_dtype);
+            }
+            push(stack, IValue(ret));
+          };
+        },
+        // TODO: need to update alias info on Tensor?
```

Comment:
do you know what's the issue here?

Reply:
Ah, sorry should have removed this todo.
The comment was put here when I don't have the alias info in the schema (since the old one copied from aten can't handle that for some reason).